vm/devices/tpm_resources/src/lib.rs

```diff
@@ -23,8 +23,8 @@ pub struct TpmDeviceHandle {
     pub nvram_store: Resource<NonVolatileStoreKind>,
     /// Whether to refresh TPM seeds on init
     pub refresh_tpm_seeds: bool,
-    /// Optional callback for getting an attestation report
-    pub get_attestation_report: Option<Resource<GetAttestationReportKind>>,
+    /// Whether the environment supports attestation
+    pub support_attestation_report: bool,
     /// Optional callback for requesting AK cert
     pub request_ak_cert: Option<Resource<RequestAkCertKind>>,
```

Comment:
These two fields can be fused into a single tri-state enum now, no? i.e: a resource-level equivalent to `TpmAkCertType`.

doing so would eliminate the need to reconstruct the tri-state enum in resolver.rs